File: 0103-binary-tree-zigzag-level-order-traversal/0103-binary-tree-zigzag-level-order-traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<IList<int>> ZigzagLevelOrder(TreeNode root) {

        Queue<TreeNode> queue=new Queue<TreeNode>();
        IList<IList<int>> ans=new List<IList<int>>();
        bool left=false;

        if(root==null)
            return ans;

        queue.Enqueue(root);

        while(queue.Count>0){
            LinkedList<int> currLevel = new LinkedList<int>();
            int Length=queue.Count;

            for(int i=0;i<Length;i++){

                TreeNode currNode = queue.Dequeue();
                
                 if (left) {
                    currLevel.AddLast(currNode.val);
                } else {
                    currLevel.AddFirst(currNode.val);
                }

                
                if(currNode.right!= null)
                    queue.Enqueue(currNode.right);
                if(currNode.left!=null)
                    queue.Enqueue(currNode.left);
                    
            }

            left=!left;
            ans.Add(new List<int> (currLevel));
        }

    
    return ans;



    }
}

File: 0103-binary-tree-zigzag-level-order-traversal/0103-binary-tree-zigzag-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque
class Solution:
    def zigzagLevelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:

        if not root:
            return []

        ans = []
        self.formTheAnswer(root,ans)
        return ans
    
    def formTheAnswer(self, node, ans):

        q = deque([node])
        left = False


        while q:
            l = []

            for _ in range(len(q)):
                if left:
                    currNode = q.pop()
                    if currNode.right:
                        q.appendleft(currNode.right)
                    if currNode.left:
                        q.appendleft(currNode.left)
                    
                    
                else:
                    currNode = q.popleft()
                    if currNode.left:
                        q.append(currNode.left)
                    if currNode.right:
                        q.append(currNode.right)

                l.append(currNode.val)

            ans.append(l)
            left = not left
        return
        


            









        